std/stdio.d

```diff
@@ -2877,13 +2877,27 @@ $(D Range) that locks the file and allows fast writing to it.
                     if (result != writeme.length) errnoEnforce(0);
                     return;
                 }
-            }
 
-            // put each element in turn.
-            alias Elem = Unqual!(ElementType!A);
-            foreach (Elem c; writeme)
+                // put each element in turn.
+                for (; !writeme.empty; writeme.popFront)
+                {
+                    put(writeme.front);
+                }
+            }
+            else static if (isSomeString!A && C.sizeof > 1)
             {
-                put(c);
+                foreach (c; writeme)
+                {
+                    put(c);
+                }
+            }
+            else
+            {
+                // put each element in turn.
+                for (; !writeme.empty; writeme.popFront)
+                {
+                    put(writeme.front);
```

Comment:
No. As pointed out in the discussion, LDC/GDC are smart enough.
The code should be improved to be more high-level and abstract, not the other way around.